### PR TITLE
GEE code adjustments, moving GEE config_files, added some documentation

### PR DESCRIPTION
**Pull request by steeleb**
Hey y'all - four items in this PR:

1) some minor adjustments to the GEE code - `b_pull_Landsat_SRST_poi/py/gee_functions.py`, `b_pull_Landsat_SRST_poi/py/runGEEperTile.py`
- dropped of some redundant code (since the image stack is now filtered to specific path/rows) - this includes some changes to other parts of the code that referenced the ee.FeatureCollection that I removed in this process (called `wrs`)
- consolidation of image masks for `ref_pull...` can be seen on lines 358-363, and then used as a single line of code in line 383. Note that this has been repeated for all 4 instances. This change is the artifact of me trying to track down the error on line 457 that should be d.eq(1) not d.eq(3), but I'm going to leave it as is, because it is more straightforward to me. 

2) addition of 'wait for task completion' step in the {targets} workflow. This is completed here: `b_pull_Landsat_SRST_poi/py/poi_wait_for_completion.py`. This is an adaptation of the `maximum_no_of_tasks` function in the `b_pull_Landsat_SRST_poi/py/gee_functions.py` file to make sure that all tasks are complete before moving on in the {targets} pipeline. While irrelevant now, the next step is to download files, so we need to make sure that all tasks are completed before doing that! (tagging ROSSyndicate/Landsat_C2_SRST#25)

3) I had to move the GEE config_files because they were getting moved around when I rendered the bookdown, which then invalidated my pipeline at the GEE pull. This is now fixed in bookdown (see [issue resolution](https://github.com/rstudio/bookdown/issues/1307)), but I'm just going to let this slide with the new config file location. This is more of an FYI for future us.

4) working on the documentation explaining the Landsat pull. This is incomplete, but a start, and the primary purpose of this PR! (Progress on #9) That said, please review the .Rmd's in the `bookdown` folder. I'm still struggling to get the vis networks to show up without the absolutely stupid message text that @mbrousil tagged in the last PR:
Screenshot 2023-10-27 at 1 52 51 PM
I'm probably going to have to add export of the visnetwork to the targets list, and then load in the .html or .png, but that's not going to be until I finish the run for this pull, so just know that I know that this is dumb. The rendered version of the bookdown is here:
[docs.zip](https://github.com/AquaSat/lakeSR/files/13192802/docs.zip)

Have an absolutely spooky weekend.

### CODE REVIEW

File: b_pull_Landsat_SRST_poi/py/runGEEperTile.py [hunk 1]
```diff
@@ -42,45 +40,45 @@
...
-wrs = (ee.FeatureCollection("projects/ee-ls-c2-srst/assets/WRS2_descending")
-  .filterMetadata("PR", "equals", tiles))
-
 # store path and row for subsetting the stacks so there is not overlap between PR pulls
 w_p = int(str(tiles)[0:3])
 w_r = int(str(tiles)[3:6])
 
 #grab images and apply scaling factors
 l7 = (ee.ImageCollection("LANDSAT/LE07/C02/T1_L2")
-    .map(apply_scale_factors)
```
Comment (steeleb) [inline b_pull_Landsat_SRST_poi/py/runGEEperTile.py:63]: Note that this is moved to the end of filtering the stack to reduce processing time.

File: b_pull_Landsat_SRST_poi/py/runGEEperTile.py [hunk 2]
```diff
@@ -42,45 +40,45 @@
...
     .filter(ee.Filter.lt("CLOUD_COVER", ee.Number.parse(str(cloud_thresh))))
     .filterDate(yml_start, yml_end)
     .filter(ee.Filter.eq("WRS_PATH", w_p))
-    .filter(ee.Filter.eq("WRS_ROW", w_r)))
+    .filter(ee.Filter.eq("WRS_ROW", w_r))
+    .map(apply_scale_factors))
     
 # merge collections by image processing groups
-ls457 = (ee.ImageCollection(l4.merge(l5).merge(l7))
```
Comment (steeleb) [inline b_pull_Landsat_SRST_poi/py/runGEEperTile.py:82]: .filterBounds is now unnecessary, as we've already filtered for path-row.

File: b_pull_Landsat_SRST_poi/py/runGEEperTile.py [hunk 3]
```diff
@@ -144,24 +142,16 @@
...
-    geo = wrs.geometry()
-    
     ## get locs feature and buffer ##
-    feat = (locs_feature
-      .filterBounds(geo)
-      .map(dp_buff))
-    
-    ## process 457 stack
-    #snip the ls data by the geometry of the location points    
-    locs_stack_ls457 = ls457.filterBounds(feat.geometry())
```
Comment (steeleb) [inline b_pull_Landsat_SRST_poi/py/runGEEperTile.py:156]: again, unnecessary, we filter for path-row.

File: b_pull_Landsat_SRST_poi/py/runGEEperTile.py [hunk 4]
```diff
@@ -144,24 +142,16 @@
   ## run the pull for LS457
   if "site" in extent:
     
-    geo = wrs.geometry()
-    
     ## get locs feature and buffer ##
-    feat = (locs_feature
```
Comment (steeleb) [inline b_pull_Landsat_SRST_poi/py/runGEEperTile.py:150]: unnecessary, features are filtered by path-row at top of script.

**Review by mbrousil — APPROVED** [review]

File: bookdown/01-Introduction.Rmd [hunk 5]
```diff
@@ -46,26 +47,40 @@ brief summary about what task(s) each group completes.
...
+artefacts from upstream products that are used to create the SR product. This
+group of targets ends with a branched target that maps over each of the WRS2
+path rows that intersect with the points. **Note**: this group of targets takes
+a very, very long time, averaging about 1 hour per path-row branch. There are
+just under 800 path rows executed in this branch.
+
+See Section \@ref(LS-C2-SRST) for additional background and detailed
+methodology.
+
+## Running the code within this repository
```
Comment (kathryn-willi) [inline bookdown/01-Introduction.Rmd:77]: this is nice

**Comment by kathryn-willi** [comment]
Thank you for sharing the rendered version of the bookdown - that was very helpful for me!

**Review by kathryn-willi — APPROVED** [review]

File: bookdown/01-Introduction.Rmd (hunk 5 above)
Comment (steeleb) [reply bookdown/01-Introduction.Rmd:77]: Thank you!

**Pull request merged**